mips/mips_tlb.sail

```diff
@@ -58,12 +58,36 @@ function tlbSearch(VAddr) =
     None()
   }
 
+val TLBSegmentOf : bits(64) -> (AccessLevel, option(bits(64))) effect pure
```

Comment:
Is this function unmodified from the original? Fine to factor it out but maybe MIPSSegmentOf would be a more appropriate name since, of course if we are being picky the TLB is not used at all for the direct-mapped regions!

Reply:
It's not modified from the original; I'd done the refactoring as part of preparing for a change that I ended up dropping.  Will rename.